src/main/java/org/truffleruby/core/string/StringNodes.java

```diff
@@ -796,12 +796,12 @@ private int countSlow(DynamicObject string, DynamicObject... otherStrings) {
         public abstract DynamicObject executeDeleteBang(VirtualFrame frame, DynamicObject string, Object[] args);
 
         @Specialization(guards = "isEmpty(string)")
-        public DynamicObject deleteBangEmpty(DynamicObject string, Object[] args) {
+        public DynamicObject deleteBangEmpty(VirtualFrame frame, DynamicObject string, Object[] args) {
```

Comment:
Wasn't the issue here the return type with `deleteBang`? I'd rather not add the `VirtualFrame` argument if we're not using it.